client/my-sites/checkout/composite-checkout/composite-checkout.js

```diff
@@ -767,6 +769,20 @@ function CountrySelectMenu( {
 	);
 }
 
+function useDetectedCountryCode() {
+	const detectedCountryCode = useSelector( getCurrentUserCountryCode );
+	const [ haveUsedDetectedCountryCode, setHaveUsedDetectedCountryCode ] = useState( false );
+
+	useEffect( () => {
```

Comment:
Using `useEffect` this way (without a dependency array) is more or less the same as just not having `useEffect` at all; its callback will be run on every render, except that it will be run just after the rendering completes. If your intent is to run this once when the component first mounts, you could use an empty dependency array. However, this has the downside in this case that if the selector returns nothing (because the state hasn't populated yet), this will have already run by the time the data is available.

I think what you might want is to say "on every render, check if we have detected a country code. Once we have one, then pass that country code along to the 'wpcom' store, after which we no longer need to check." For that last part, a `useRef` might be sufficient rather than a `useState` since the latter causes an additional re-render (which should already by handled by the change of the 'wpcom' store).

Reply:
Good catch. I forget about the dependency array every time.